ops/testing.py

```diff
@@ -1141,6 +1141,67 @@ def reset_planned_units(self):
         """
         self._backend._planned_units = None
 
+    def add_network(self, binding_name: str, address: str, *,
+                    relation_id: Optional[int] = None,
+                    cidr: Optional[str] = None,
+                    interface: str = 'eth0',
+                    ingress_addresses: Optional[List[str]] = None,
+                    egress_subnets: Optional[List[str]] = None,
+                    ):
+        """Add simulated network data for the given binding.
+
+        Calling this multiple times with the same (binding_name, relation_id)
+        combination will replace the associated network data.
+
+        This method only supports the common case of adding a single interface
+        and bind address. If you need support for multiple bind addresses, you
+        can patch ops.testing._TestingModelBackend.network_get and return the
```

Comment:
`_TestingModelBackend` seems to be private, is it advisable to patch that? In some ways that becomes part of the API that would result in breaking changes for users which might not be what is intended for a private class?

Reply:
No, it's not advisable to patch that -- but that's what people have been doing to date, so hopefully `add_network` gets rid of the need for that entirely. :-) I've just dropped that sentence. People will still hack stuff if they really need the complexity, but hopefully just using `add_network` is good enough.